alpaca/broker/models/requests.py

```diff
@@ -314,3 +319,138 @@ def root_validator(cls, values: dict) -> dict:
 
         return values
+
+
+class GetTradeDocumentsRequest(NonEmptyRequest):
+    """
+    Represents the various filters you can specify when making a call to get TradeDocuments for an Account
+
+    Attributes:
+        start (Optional[Union[date, str]]): Filter to TradeDocuments created after this Date. str values will attempt to
+          be upcast into date instances. Format must be in YYYY-MM-DD.
+        end (Optional[Union[date, str]]): Filter to TradeDocuments created before this Date. str values will attempt to
+          be upcast into date instances. Format must be in YYYY-MM-DD.
+        type (Optional[TradeDocumentType]): Filter to only these types of TradeDocuments
+    """
+
+    start: Optional[Union[date, str]] = None
+    end: Optional[Union[date, str]] = None
+    type: Optional[TradeDocumentType] = None
+
+    def __init__(self, **data) -> None:
+        if "start" in data and isinstance(data["start"], str):
+            data["start"] = date.fromisoformat(data["start"])
+
+        if "end" in data and isinstance(data["end"], str):
+            data["end"] = date.fromisoformat(data["end"])
+
+        super().__init__(**data)
+
+    @root_validator()
+    def root_validator(cls, values: dict) -> dict:
+        if (
+            "start" in values
+            and values["start"] is not None
+            and "end" in values
+            and values["end"] is not None
+            and values["start"] > values["end"]
+        ):
+            raise ValueError("start must not be after end!!")
+
+        return values
+
+
+class UploadDocumentRequest(NonEmptyRequest):
+    """
+    Attributes:
+        document_type (UploadDocumentType): The type of document you are uploading
+        document_sub_type (Optional[UploadDocumentSubType]): If supported for the corresponding `document_type` this
+          field allows you to specify a sub type to be even more specific.
+        content (str): A string containing Base64 encoded data to upload.
+        mime_type (UploadDocumentMimeType): The mime type of the data in `content`
+    """
+
+    document_type: UploadDocumentType
+    document_sub_type: Optional[UploadDocumentSubType] = None
+    content: str
+    mime_type: UploadDocumentMimeType
+
+    @root_validator()
+    def root_validator(cls, values: dict) -> dict:
+        if values["document_type"] == UploadDocumentType.W8BEN:
+            raise ValueError(
+                "Error please use the UploadW8BenDocument class for uploading W8BEN documents"
+            )
+
+        if values["document_sub_type"] == UploadDocumentSubType.FORM_W8_BEN:
+            raise ValueError(
+                "Error please use the UploadW8BenDocument class for uploading W8BEN documents"
+            )
+
+        return values
+
+
+class UploadW8BenDocumentRequest(NonEmptyRequest):
+    """
+    Attributes:
+        document_type (UploadDocumentType, optional): Must be UploadDocumentType.W8BEN. Will default to W8BEN if not set
```

Comment:
`document_type`  is optional in the docstrings but not in the class properties.

Reply:
Yeah I wasn't sure how to mark that they'll get defaulted in the constructor to a value. They aren't marked optional in the class properties because they can't be anything other than the W8BEN types. Having them be optional in the type would imply you could set them to None after initialization :thinking:. 

I think we can just remove them from the doc string and make them just auto set in the constructor but not have them be real properties. The root validator should still validate em if people try to set them manually for some reason :thinking:

Reply:
I've gone ahead with what I said in the above message there; let me know if you have another option for this

Reply:
Aah I see what you mean. I like your solution.